### PR TITLE
Core 573

### PR DESCRIPTION
**Pull request by sri2k1us**
- This is on top of Apps Listing PR. So reviewing last [commit](https://github.com/cyverse-de/ui/pull/455/commits/1b54fd8a3d4e3942c4dd6d5eab4a1f8b80502300) is enough.

image

### CODE REVIEW

File: react-components/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
         "prop-types": "^15.7.2"
     },
     "dependencies": {
-        "@cyverse-de/ui-lib": "^0.3.8",
+        "@cyverse-de/ui-lib": "file:../../ui-lib/cyverse-de-ui-lib-0.3.8.tgz",
```
Comment (ash3rz) [inline react-components/package.json]: Reminder to change before merging 😉

File: react-components/src/apps/details/AppInfoDialog.js [hunk 2]
```diff
@@ -33,6 +33,13 @@ import Tab from "@material-ui/core/Tab";
 export const EDIT_MODE = "edit";
 export const VIEW_MODE = "view";
 
+const TAB_INDEX = {
+    APP_DETAIL_TAB: 0,
+    QUICK_LAUNCH_TAB: 1,
+    TOOL_DETAILS_TAB: 2,
+    APP_DOC: 3,
```
Comment (ash3rz) [inline react-components/src/apps/details/AppInfoDialog.js]: 👍

**Comment by sri2k1us** [comment]
> LGTM 🚢
> Unrelated to this PR though, is the App dot menu the same as the Analysis dot menu? The padding/spacing between the menu items seems more condensed

Good Catch. Removed the padding that i had added and that fixed it.

File: react-components/src/apps/listing/AppGridListing.js [hunk 3]
```diff
@@ -243,8 +251,9 @@ function getTableColumns(deletable, enableMenu) {
         {
             name: AppFields.RATING.fieldName,
             enableSorting: true,
-            key: AppFields.RATING.key,
+            key: "rating",
```
Comment (psarando) [inline react-components/src/apps/listing/AppGridListing.js]: Another rebase mishap?

File: de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java [hunk 4]
```diff
@@ -26,6 +26,8 @@
     private final App app;
     private final AppDetailsAppearance appearance;
     private String searchText;
+    private String searchRegexPattern;
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java]: Oops...I made a Boo Boo 😖

File: react-components/src/apps/listing/AppGridListing.js (hunk 3 above)
Comment (sri2k1us) [reply react-components/src/apps/listing/AppGridListing.js]: Another one 🤦‍♂

File: de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (psarando) [reply de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java]: Should this line be removed now?

[the remark this answers is not in the document]
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java]: Yes, that line can be removed. Not sure why the rebase didn't catch that.

**Comment by sri2k1us** [comment]
Thanks for the reviews. And @psarando 🦅 👀

**Pull request merged**